### PR TITLE
Unr 4845 move sender

### PR DESCRIPTION
**Pull request by jacquese**
#### Description
Generous amounts of copy pasting from one class (sender) to another (actor system). Removes some dependencies on static component view, but not all of them because I don't want to rabbit hole again.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp [hunk 1]
```diff
@@ -256,7 +256,7 @@ void USpatialActorChannel::RetireEntityIfAuthoritative()
 			}
 			else
 			{
-				Sender->RetireEntity(EntityId, Actor->IsNetStartupActor());
+				NetDriver->ActorSystem->RetireEntity(EntityId, Actor->IsNetStartupActor());
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp:259]: Lots of removal of sender from actor channel, can we remove the member variable now also?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp [hunk 2]
```diff
@@ -1280,7 +1280,18 @@ void USpatialActorChannel::SendPositionUpdate(AActor* InActor, Worker_EntityId I
...
 	if (InEntityId != SpatialConstants::INVALID_ENTITY_ID && NetDriver->HasServerAuthority(InEntityId))
 	{
-		Sender->SendPositionUpdate(InEntityId, NewPosition);
+#if !UE_BUILD_SHIPPING
+		if (!NetDriver->Connection->GetView()[EntityId].Authority.Contains(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID))
+		{
+			UE_LOG(LogSpatialActorChannel, Verbose,
+				   TEXT("Trying to send Position component update but don't have authority! Update will not be sent. Entity: %lld"),
+				   EntityId);
+			return;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp:1289]: NetDriver->HasServerAuthority checks this already, why do it again?

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp:1289]: blindly moved what was inside sender to here. Thanks for calling out

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp (hunk 1 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp:259]: not yet, need to deal with RPCs etc in upcoming PRs

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 3]
```diff
@@ -835,8 +834,8 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 	if (IsServer())
 	{
 		if (GlobalStateManager != nullptr && !GlobalStateManager->GetCanBeginPlay()
-			&& StaticComponentView->HasAuthority(GlobalStateManager->GlobalStateManagerEntityId,
-												 SpatialConstants::GDK_KNOWN_ENTITY_AUTH_COMPONENT_SET_ID))
+			&& Connection->GetCoordinator().HasAuthority(GlobalStateManager->GlobalStateManagerEntityId,
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:837]: Wonder if this should be bespoke functionality on the well-known-entities system?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 4]
```diff
@@ -2330,12 +2329,12 @@ bool USpatialNetDriver::CreateSpatialNetConnection(const FURL& InUrl, const FUni
 
 bool USpatialNetDriver::HasServerAuthority(Worker_EntityId EntityId) const
 {
-	return StaticComponentView->HasAuthority(EntityId, SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:2333]: Static component view used at all anymore? If not, what's left?

File: SpatialGDK/Source/SpatialGDK/Public/Interop/ActorSystem.h [hunk 5]
```diff
@@ -69,6 +93,7 @@ class ActorSystem
 	void AuthorityLost(Worker_EntityId EntityId, Worker_ComponentSetId ComponentSetId);
 	void AuthorityGained(Worker_EntityId EntityId, Worker_ComponentSetId ComponentSetId);
 	void HandleActorAuthority(Worker_EntityId EntityId, Worker_ComponentSetId ComponentSetId, Worker_Authority Authority);
+	void ProcessUpdatesQueuedUntilAuthority(Worker_EntityId EntityId);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/ActorSystem.h:96]: Remove old function

File: SpatialGDK/Source/SpatialGDK/Public/SpatialView/ViewCoordinator.h [hunk 6]
```diff
@@ -94,6 +94,10 @@ class ViewCoordinator
 		Worker_ComponentId ComponentId,
 		const FAuthorityChangeRefreshPredicate& RefreshPredicate = FSubView::NoAuthorityChangeRefreshPredicate);
 
+	bool HasEntity(Worker_EntityId EntityId) const;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/SpatialView/ViewCoordinator.h:97]: nice

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp [hunk 7]
```diff
@@ -48,52 +48,6 @@ DECLARE_CYCLE_STAT(TEXT("Sender QueueOutgoingUpdate"), STAT_SpatialSenderQueueOu
 DECLARE_CYCLE_STAT(TEXT("Sender UpdateInterestComponent"), STAT_SpatialSenderUpdateInterestComponent, STATGROUP_SpatialNet);
 DECLARE_CYCLE_STAT(TEXT("Sender SendRPC"), STAT_SpatialSenderSendRPC, STATGROUP_SpatialNet);
 
-namespace
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp:51]: Instead of going through this and ActorSystem.cpp, can you call out any changes that aren't direct copies?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 3 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:837]: the has auth check? or the whole block? The block probably should, actually. Will move

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:837]: The `DoYouHaveAuthorityOverGSM` function

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:837]: moved the ifserver part into well known entity system

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 4 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:2333]: yeah used in a couple places, the actor channel being one of them. Need to add net owning client worker back to the actor system's data store for that one, which was a little more involved. There's another ticket for dealing with the remaining scv bits: https://improbableio.atlassian.net/browse/UNR-4844

File: SpatialGDK/Source/SpatialGDK/Public/Interop/ActorSystem.h (hunk 5 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Public/Interop/ActorSystem.h:96]: gone

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp (hunk 7 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp:51]: had a scan- the only places that weren't direct copies were the stuff in spatial net driver with not updating SCV any more, and the logic around queuing updates until auth, so nothing left in actor system and sender

File: SpatialGDK/Source/SpatialGDK/Private/Interop/WellKnownEntitySystem.cpp [hunk 8]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/WellKnownEntitySystem.cpp:1]: new line

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 9]
```diff
@@ -890,8 +880,8 @@ void USpatialNetDriver::SpatialProcessServerTravel(const FString& URL, bool bAbs
 	UWorld* World = GameMode->GetWorld();
 	USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(World->GetNetDriver());
 
-	if (!NetDriver->StaticComponentView->HasAuthority(SpatialConstants::INITIAL_GLOBAL_STATE_MANAGER_ENTITY_ID,
-													  SpatialConstants::GDK_KNOWN_ENTITY_AUTH_COMPONENT_SET_ID))
+	if (!NetDriver->Connection->GetCoordinator().HasAuthority(SpatialConstants::INITIAL_GLOBAL_STATE_MANAGER_ENTITY_ID,
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:883]: Driveby - but seems we should be using `GlobalStateManager->GlobalStateManagerEntityId` here instead.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp (hunk 7 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp:51]: Gave it a scan - looks good

**Review by m-samiec — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDK/Private/Interop/WellKnownEntitySystem.cpp (hunk 8 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/WellKnownEntitySystem.cpp:1]: dealt with

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 9 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: done
